### PR TITLE
Support read operation of binary resource

### LINKED ISSUE #42: CBOR data type support in Virtual Device
We need to make the CBOR support in Virtual Device Service for demonstrate how the binary readings work.

### PR DESCRIPTION
**Pull request by FelixTing**
close #42 

Signed-off-by: Felix Ting <felix@iotechsys.com>

### CODE REVIEW

**Comment by FelixTing** [comment]
wait for [edgex-go #2272](https://github.com/edgexfoundry/edgex-go/issues/2272) to be resolved

File: internal/driver/virtualdriver.go [hunk 1]
```diff
@@ -234,6 +237,12 @@ func prepareVirtualResources(driver *VirtualDriver, deviceName string) error {
 						dr.Properties.Value.Type <-> VIRTUAL_RESOURCE.DataType
 						dr.Properties.Value.DefaultValue <-> VIRTUAL_RESOURCE.Value
 					*/
+					if dsModels.ParseValueType(dr.Properties.Value.Type) == dsModels.Binary {
+						b := make([]byte, dsModels.MaxBinaryBytes)
+						rand.Seed(time.Now().UnixNano())
+						rand.Read(b)
+						dr.Properties.Value.DefaultValue = hex.EncodeToString(b)
```
Comment (cloudxxx8) [inline internal/driver/virtualdriver.go]: Why are you using dr.Properties.Value.DefaultValue instead of a new variable here?

**Comment by cloudxxx8** [comment]
please handle the check error

> [2020-02-27T06:30:44.281Z] + make test
> [2020-02-27T06:30:44.281Z] CGO_ENABLED=0 GO111MODULE=on go test ./... -coverprofile=coverage.out
> [2020-02-27T06:30:46.192Z] ?   	github.com/edgexfoundry/device-virtual-go	[no test files]
> [2020-02-27T06:31:04.370Z] ?   	github.com/edgexfoundry/device-virtual-go/cmd	[no test files]
> [2020-02-27T06:35:56.150Z] ok  	github.com/edgexfoundry/device-virtual-go/internal/driver	278.583s	coverage: 51.3% of statements
> [2020-02-27T06:35:56.150Z] CGO_ENABLED=0 GO111MODULE=on go vet ./...
> [2020-02-27T06:35:56.150Z] gofmt -l .
> [2020-02-27T06:35:56.150Z] [ "`gofmt -l .`" = "" ]
> [2020-02-27T06:35:56.150Z] ./bin/test-attribution-txt.sh
> [2020-02-27T06:35:56.150Z] An attribution for github.com/OneOfOne/xxhash is missing from in Attribution.txt, please add
> [2020-02-27T06:35:56.150Z] An attribution for github.com/jessevdk/go-flags is missing from in Attribution.txt, please add
> [2020-02-27T06:35:56.150Z] make: *** [Makefile:27: test] Error 1
> script returned exit code 2

**Comment by FelixTing** [comment]
re-create this PR using personal GitHub account

**Pull request closed without merging**